### PR TITLE
Merge hotfix 3.4.2 into develop

### PR DESCRIPTION
**Pull request by thomaslevesque**
Supersedes #1142

### CODE REVIEW

**Comment by blairconrad** [comment]
I'm surprised to see "[Merge pull request #1130 from FakeItEasy/hotfix-3.4.1  …](https://github.com/FakeItEasy/FakeItEasy/pull/1143/commits/f2d27a24531078dc0fc2220c84fdb852122adacb)" in the commit list.
I guess it makes sense, though…

**Comment by thomaslevesque** [comment]
> I'm surprised to see "Merge pull request #1130 from FakeItEasy/hotfix-3.4.1 …" in the commit list.

That's because we didn't merge hotfix-3.4.1 into develop (because there was nothing to merge)

**Comment by blairconrad** [comment]
> That's because we didn't merge hotfix-3.4.1 into develop (because there was nothing to merge)

fca1bf0e3d2a1676c748fff4d710f6b024fb7448

**Comment by blairconrad** [comment]
I think it's because while we _did_ merge hotfix-3.4.1 into develop, we also merged into master, and that's what got tagged, so the merge to master is the commit that we're no merging, and it's a no-op because the content made it into develop already.

I think Git is doing the right thing. I just think it's confusing.

**Comment by blairconrad** [comment]
On that note, if we merge your PR, and then later release 3.4.3, I think when we try to merge hotfix-3.4.3 into develop, the missing commit will come back!

**Comment by thomaslevesque** [comment]
> On that note, if we merge your PR, and then later release 3.4.3, I think when we try to merge hotfix-3.4.3 into develop, the missing commit will come back!

Oh... I think you're right. I wanted to leave no trace of this commit, which is why I did it like this. OK, let me try again, then.

**Comment by blairconrad** [comment]
The branches are getting wide, but I guess this is the best we can do if we want to use GitHub to merge:
image

I still think we're going to see the missing commit come back if we make a 3.4.3, but I do not know how we can possibly avoid that.

Shall I go ahead, then?

**Comment by thomaslevesque** [comment]
> I still think we're going to see the missing commit come back if we make a 3.4.3, but I do not know how we can possibly avoid that.

Why? It's part of this merge, so it shouldn't come back later.

> Shall I go ahead, then?

👍

**Comment by blairconrad** [comment]
> Why? It's part of this merge, so it shouldn't come back later.

Ah, yes. I see. Sorry.

**Comment by thomaslevesque** [comment]
Thanks for the merge!
